README.md

```diff
@@ -1,4 +1,241 @@
+
+#### How it works
+
+So, our `#save` method inserts a record into our database that has the name and album values of the song instance we are trying to save. We are not saving the Ruby object itself. We are creating a new row in our songs table that has the values that characterize that song instance. 
+
+**Important:** Notice that we *didn't* insert an ID number into the table with the above statement. Remember that the `INTEGER PRIMARY KEY` datatype will assign and auto-increment the id attribute of each record that gets saved.
+
+## Creating Instances vs. Creating Table Rows
+
+The moment in which we create a new `Song` instance with the `#new` method is *different than the moment in which we save a representation of that song to our database*. The `#new` method creates a new instance of the song class, a new Ruby object. The `#save` method takes the attributes that characterize a given song and save it to the songs table in our database as its own table row. 
+
+We can make our own decisions about when and where to call the `#save` method. You could choose to call that method *inside the initialize method*, to automatically save a song's information to the database as soon as that new song instance is created. Or, we could collect all of our song instances in an class variable, `@@all`, and save the members of that collection at a later time. 
+
+Let's take a look at that option now:
+
+```ruby
+class Song
+  
+  @@all = []
+
+  attr_accessor :name, :album
+  
+  attr_reader :id
+  
+  def initialize(name, album, id=nil)
+    @id = id
+    @name = name
+    @album = album
+    @@all << self
+  end
+  
+  def self.create_table
+    sql =  <<-SQL 
+      CREATE TABLE IF NOT EXISTS songs (
+        id PRIMARY KEY INTEGER, 
+        name TEXT, 
+        album TEXT
+        )
+        SQL
+    DB[:conn].execute(sql) 
+  end
+  
+  def save
+    sql = <<-SQL
+      INSERT INTO songs (name, album) 
+      VALUES (?, ?)
+    SQL
+    
+    DB[:conn].execute(sql, self.name, self.album)
+    
+  end
+  
+  def self.all
+    @@all
+  end
+
+end
+```
+
+Now, we can create and save songs like this:
+
+```ruby
+Song.create_table
+Song.new("Hello", "25")
+Song.new("99 Problems", "The Black Album")
+
+Song.all.each do |song|
+  song.save
+end
+```
+
+Here we:
+
+* Create the songs table. 
+* Create two new song instances. 
+* Iterate over our collection of song instances stored in `Song.all` and use the `song.save` method to persist them to the database. 
+
+## Conclusion
+
+The important concept to grasp here, and it's not easy, is the idea that we are *not* saving Ruby objects into our database. We are using the attributes of a given Ruby object to create a new row in our database table. 
```

Comment:
Before you conclude point out that if you ahve the same instance and save twice, it creates two entries in the DB. We'll fix that in future labs!